services/ui-src/src/components/menus/Sidebar.tsx

```diff
@@ -0,0 +1,224 @@
+import NavItems from "data/navigation/MCPARSideNavItems";
+
+interface LinkItemProps {
+  name: string;
+  path: string;
+  children?: LinkItemProps[] | null;
+}
+
+const basePath = "/mcpar";
+
+export const Sidebar = () => {
+  const { pathname } = useLocation();
+  const isMcparReport = pathname.includes("/mcpar");
+  const [isOpen, setIsOpen] = useState(true);
+  return (
+    <>
+      {isMcparReport && (
+        <Box sx={sx.root} className={isOpen ? "open" : "closed"}>
+          <Box
+            as="button"
+            sx={sx.closeButton}
+            onClick={() => setIsOpen(!isOpen)}
+          >
+            <ArrowIcon
+              title="closeNavBarButton"
+              direction={isOpen ? "left" : "right"}
+            />
+          </Box>
+          <Box sx={sx.topBox}>
+            <Heading sx={sx.title}>MCPAR Report Submission Form</Heading>
+          </Box>
+          {NavItems.map((section) => (
+            <NavSection
+              key={section.name}
+              section={section}
+              level={1}
+              basePath={basePath}
+            />
+          ))}
+        </Box>
+      )}
+    </>
+  );
+};
+
+interface NavSectionProps {
+  key: string;
+  section: LinkItemProps;
+  level: number;
+  basePath: string;
+}
+
+const NavSection = ({ section, level, basePath }: NavSectionProps) => {
+  const [isOpen, setIsOpen] = useState(false);
+  const { name, path, children } = section;
+  const itemPath = `${basePath}${path}`;
+  return (
+    <>
+      <React.Fragment key={itemPath}>
+        {children ? (
+          <Box as="button" onClick={() => setIsOpen(!isOpen)}>
+            <NavItem
+              name={name}
+              level={level}
+              optionPath={itemPath}
+              hasChildren={!!children}
+              isOpen={isOpen}
+            />
+          </Box>
+        ) : (
+          <Link as={RouterLink} to={itemPath}>
+            <NavItem
+              name={name}
+              level={level}
+              optionPath={itemPath}
+              hasChildren={!!children}
+            />
+          </Link>
+        )}
+        {!!children && (
+          <Collapse in={isOpen}>
+            {children.map((section) => (
+              <NavSection
+                key={section.name}
+                section={section}
+                level={level + 1}
+                basePath={itemPath}
+              />
+            ))}
+          </Collapse>
+        )}
+      </React.Fragment>
+    </>
+  );
+};
+
+interface NavItemProps {
+  name: string;
+  optionPath: string;
+  level: number;
+  hasChildren: boolean;
+  isOpen?: boolean;
+}
+
+const NavItem = ({
+  name,
+  optionPath,
+  level,
+  hasChildren,
+  isOpen,
+}: NavItemProps) => {
+  const currentPath = window.location.pathname;
+  const selectedClass = optionPath === currentPath ? "selected" : "";
+  return (
+    <Flex sx={sx.navItemFlex} className={selectedClass}>
```

Comment:
Nit: I'd maybe call this `isCurrentPath` or something, but nothing wrong with this either.

Reply:
changed